File: java/Array/SignRearrangement.java
import java.util.ArrayList;
import java.util.Scanner;
public class SignRearrangement {

    public static ArrayList<Integer> isRearrange(ArrayList<Integer> arr,int n){
        // ArrayList<Integer>pos=new ArrayList<>();
        // ArrayList<Integer>neg=new ArrayList<>();

        // for(int i=0;i<n;i++)
        // {
        //     if(arr.get(i)>0) pos.add(arr.get(i));
        //     else neg.add(arr.get(i));
        // }

        // for(int i=0;i<n/2;i++)
        // {
        //     arr.set(i*2,pos.get(i));
        //     arr.set(i*2+1,neg.get(i));
        // }
        // return arr;

        ArrayList<Integer>ans=new ArrayList<Integer>(n);
        
        int pos=0,neg=1;
        for(int i=0;i<n;i++)
        {
            if(arr.get(i)<0)
            {
                ans.set(neg,arr.get(i));
                neg+=2;
        }
        else{
            ans.set(pos,arr.get(i));
                pos+=2;
            }
        }
        return ans;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        ArrayList<Integer> arr=new ArrayList<>();
        for(int i=0;i<n;i++){
            arr.add(sc.nextInt());
        }
        System.out.println(isRearrange(arr,n));
        sc.close();
    }   
}

File: cpp/Array/SignRearrangement.cpp
#include<bits/stdc++.h>
using namespace std;
vector<int> signRearrangement(vector<int> a,int n) {
    // vector<int>pos;
    // vector<int>neg;

    // for(int i=0;i<n;i++)
    // {
    //     if(a[i]>0) pos.push_back(a[i]);
    //     else neg.push_back(a[i]);
    // }

    // for(int i=0;i<n/2;i++)
    // {
    //     a[i*2]=pos[i];
    //     a[i*2+1]=neg[i];
    // }
    // return a;

    vector<int>ans(n,0);

    int pos=0,neg=1;
    for(int i=0;i<n;i++)
    {
        if(a[i]<0)
        {
            ans[neg]=a[i];
            neg+=2;
        }
        else{
            ans[pos]=a[i];
            pos+=2;
        }
    }
    return ans;
}
int main()
{
    int n;
    cin>>n;
    vector<int>a(n);
    for(int i=0;i<n;i++)
    {
        cin>>a[i];
    }

    vector<int>ans=signRearrangement(a,n);
    for(int i=0;i<n;i++)
    {
        cout<<ans[i]<<" ";
    }
    return 0;
}

//TC=O(n+n/2) SC=O(n)
//TC=O(n) SC=O(n)